Backend/services/userService.js

```diff
@@ -0,0 +1,51 @@
+const mongoose = require("mongoose");
+const User = require("../models/userModel");
+const hashPassword = require("../utils/hashPassword");
+
+const checkExistingUser = async (email) => {
+  return await User.findOne({ email });
```

Comment:
it is better to return a new Promise from a service function and then await for it when calling in a controller.
If you dont want to return a new Promise then, you can omit await keyword inside service function because you are alreading awaiting this function when calling inside controller.